File: guides/micronaut-data-json-duality-view/java/src/main/java/example/micronaut/domain/StudentClass.java
package example.micronaut.domain;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.MappedProperty;
import io.micronaut.data.annotation.Relation;

@MappedEntity(value = "TBL_STUDENT_CLASSES", alias = "sc") // <1>
public record StudentClass (
        @Id // <2>
        @GeneratedValue(GeneratedValue.Type.IDENTITY) // <3>
        Long id,
        @Relation(Relation.Kind.MANY_TO_ONE) // <4>
        Student student,
        @Relation(Relation.Kind.MANY_TO_ONE) // <4>
        @MappedProperty("CLASS_ID") // <5>
        Class clazz
) {}

File: guides/micronaut-data-json-duality-view/kotlin/src/main/kotlin/example/micronaut/domain/StudentClass.kt
package example.micronaut.domain

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.MappedProperty
import io.micronaut.data.annotation.Relation

@MappedEntity(value = "TBL_STUDENT_CLASSES", alias = "sc")
data class StudentClass(
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    val id: Long?,
    @Relation(Relation.Kind.MANY_TO_ONE)
    val student: Student,
    @Relation(Relation.Kind.MANY_TO_ONE)
    @MappedProperty("CLASS_ID")
    val clazz: Class
)
